File: pset6/caesar.py
import cs50
import sys

def main():
   if(len(sys.argv) != 2):
       print("Usage: ./caesar k")
       exit(1)
   else:
       converted = int(sys.argv[1])
       key = 0
       if(converted < 0):
           print("Key must be a non-negative integer.")
           exit(1)
       elif(converted >= 26):
           key = converted - 26
       else:
           key = converted
       print("plaintext: ", end="")
       text = cs50.get_string()
       print("ciphertext: ", end="")
       for i in range(len(text)):
           if not (text[i].isalpha()):
               print("{}".format(text[i]), end="")
               continue
           alphabeticalIndex = ord(text[i].upper()) - 65
           newAlphabeticalIndex = int((alphabeticalIndex + key) % 26)
           asciiIndex = newAlphabeticalIndex + 65
           asciiLetter = chr(asciiIndex)
           if(text[i].isupper()):
               print("{}".format(asciiLetter.upper()), end="")
           else:
               print("{}".format(asciiLetter.lower()), end="")
       print("")
       

if __name__ == "__main__":
    main()

File: pset2/caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>


int main(int argc, string argv[])
{
    if(argc != 2) {
        printf("Usage: ./caesar k\n");
        return 1;
    }
    else
    {
        //convert string to int
        int converted = atoi(argv[1]);      
        int key = 0;
        if(converted < 0) {
            printf("Key must be a non-negative integer.\n");
            return 1;
        }
        else if(converted >= 26){
            //in order to stay in the range of 0 to 26
            key = converted - 26;
        } else {
            key = converted;
        }
        printf("plaintext: ");
        string text = get_string();
        printf("ciphertext: ");
        for(int i = 0, n = strlen(text); i < n; i++)
        {
            //in case it is not an alphabetical character it should not be changed
            if(isalpha(text[i]) == false){
                printf("%c", text[i]);
                continue;
            }
            //get the alphabetical index in order to use the below formula
            int alphabeticalIndex = toupper(text[i]) - 65;
            int newAlphabeticalIndex = (alphabeticalIndex + key) % 26;
            //convert back to ascii by adding 65 
            int asciiIndex = newAlphabeticalIndex + 65;
            
            //the case should be preserved
            if(isupper(text[i])){
                printf("%c", toupper(asciiIndex));
            } else{
                printf("%c", tolower(asciiIndex));
            }
        }
        printf("\n");
    }
}